Makefile

```diff
@@ -1,53 +1,110 @@
-
-APPVERSION := latest
-GOVERSION := 1.21
-PROGRAM := pg_tileserv
-CONFIG := config/$(PROGRAM).toml
-CONTAINER := pramsey/$(PROGRAM)
-DATE := $(shell date +%Y%m%d)
+##AVAILABLE BUILD OPTIONS -
+##      APPVERSION - Variable to set the version label
+##      GOVERSION - Defaults to 1.21.6 but can be overriden, uses alpine go container as base
+##      PROGRAM - Name of binary, pg_tileserv
+##      CONTAINER - prefix and name of the generated container
+##      CONFIG - config file to be used
+##      DATE - Date String used as alternate tag for generated containers
+##      BASE_REGISTRY - This is the registry to pull the base image from
+##      BASE_IMAGE - The base image to use for the final container
+##      TARGETARCH - The architecture the resulting image is based on and the binary is compiled for
+##      IMAGE_TAG - The container and tag to be applied to the container
+
+APPVERSION ?= latest
+GOVERSION ?= 1.21.6
+PROGRAM ?= pg_tileserv
+CONFIG ?= config/$(PROGRAM).toml
+CONTAINER ?= pramsey/$(PROGRAM)
+DATE ?= $(shell date +%Y%m%d)
+BASE_REGISTRY ?= registry.access.redhat.com
+BASE_IMAGE ?= ubi8-micro
+SYSTEMARCH = $(shell uname -i)
+
+ifeq ($(SYSTEMARCH), x86_64)
+TARGETARCH ?= amd64
+PLATFORM=amd64
+else
+TARGETARCH ?= arm64
+PLATFORM=arm64
+endif
+
+IMAGE_TAG ?= $(CONTAINER):$(APPVERSION)-$(TARGETARCH)
 
 RM = /bin/rm
 CP = /bin/cp
 MKDIR = /bin/mkdir
 
-.PHONY: bin-docker build-docker build check clean docs install release test uninstall
+.PHONY: bin-docker bin-for-docker build build-docker check clean common-build docker docs install multi-stage-docker release set-local set-multi-stage test uninstall
 
 .DEFAULT_GOAL := help
 
 GOFILES := $(wildcard *.go)
 
 all: $(PROGRAM)
 
-check:  ##         This checks the current version of Go installed locally
+check:  ##              This checks the current version of Go installed locally
 	@go version
 
-clean:  ##         This will clean all local build artifacts
+clean:  ##              This will clean all local build artifacts
 	$(info Cleaning project...)
 	@rm -f $(PROGRAM)
 	@rm -rf docs/*
-	docker image prune --force
+	@docker image inspect $(CONTAINER):$(APPVERSION) >/dev/null 2>&1 && docker rmi -f $(CONTAINER):$(APPVERSION) $(CONTAINER):$(DATE) || echo -n ""
```

Comment:
Is this working? My make clean is not removing the build artefacts or utility images.